all_dash_public/all-dash-public.py

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python
+import argparse
+import os
+import sys
+import textwrap
+from datadog import initialize
+from datadog import api
+import json
+from halo import Halo
+
+def getAllPublicDashboards():
+    # @ckelner: These won't work - no way to get public token from API at this time
+    # result = api.Dashboard.get_all()
+    # result = api.Screenboard.get_all()
+    # result = api.Screenboard.get(<redacted>)
+    # result = api.Dashboard.get(<redacted>)
+    spinner = Halo(text="Getting all public dashboards; This might take awhile...", spinner="dots")
+    spinner.start()
+
+    # dict to hold the public dashboard info
+    public_dashboards = {}
+    # get all dashboard lists
+    d_lists = api.DashboardList.get_all()["dashboard_lists"]
+    # iterate over the lists
+    for list in d_lists:
+        # ignore those with no dashboards
+        if list["dashboard_count"] > 0:
+            # get each dashboard
+            dashboards = api.DashboardList.get_items(list["id"])["dashboards"]
```

Comment:
you are n squaring by iterating over dashboards within the lists iterator.  iterate through the lists and add the dashboards to a separate list 'dashboards'.  then iterate over those dashboards and add them to pubic dashboards.

Reply:
👋 Hey @burnsie7 thanks for review -- updated.

Went from `19.140s` to `13.170s` 8) (at least against `11287`)